test/e2e/etcdctl_test.go

```diff
@@ -0,0 +1,116 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/openshift/cluster-etcd-operator/test/e2e/framework"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TestEtcdctlCommands executes all known etcdctl commands inside of the etcdctl container.
+// The test is not intended to be a functional test yet a sanity test that the container
+// ENV is populated correctly and that etcdctl consumes that ENV properly.
+func TestEtcdctlCommands(t *testing.T) {
+	// setup
+	cs := framework.NewClientSet("")
+	pod, err := cs.CoreV1Interface.Pods("").List(context.TODO(), metav1.ListOptions{LabelSelector: "k8s-app=etcd"})
+	require.NoError(t, err)
+	podName := pod.Items[0].Name
+
+	testCases := []struct {
+		command       string
+		skip          bool
+		skipReason    string
+		expectedError string
+	}{
+		{"etcdctl alarm disarm", false, "", ""},
+		{"etcdctl alarm list", false, "", ""},
+		{"etcdctl del --prefix /etcdctl-check-perf/", false, "", ""},
+		{"etcdctl check perf", false, "", ""},
+		{"etcdctl compaction 0", false, "", "Error: etcdserver: mvcc: required revision has been compacted"},
+		{"etcdctl defrag", false, "", ""},
```

Comment:
Is this potentially disruptive?

Reply:
it could be now but it should run only against localhost in which case it should not.

Reply:
the fact that it runs against all members now is a bug we are looking at now.